client/my-sites/plans-comparison/is-eligible-for-pro-plan.ts

```diff
@@ -1,17 +1,25 @@
 import { isEnabled } from '@automattic/calypso-config';
+import { isPro } from '@automattic/calypso-products';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import isSiteWPForTeams from 'calypso/state/selectors/is-site-wpforteams';
-import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { isJetpackSite, getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import type { AppState } from 'calypso/types';
 
-export function isEligibleForProPlan( state: AppState, siteId?: number ): boolean {
+export function isEligibleForProPlan( state: AppState, siteId?: number | null ): boolean {
 	if (
 		siteId &&
 		( ( isJetpackSite( state, siteId ) && ! isAtomicSite( state, siteId ) ) ||
 			isSiteWPForTeams( state, siteId ) )
 	) {
 		return false;
 	}
+	siteId = siteId || getSelectedSiteId( state );
+	const selectedSite = getSite( state, siteId || '' );
```

Comment:
Is `siteId || ''` really needed? And if `''` is used for the siteId would the `selectedSite` return what we expect to?

Reply:
This was needed to reconcile the type differences of the optional `siteId` argument;  `isJetpackSite`, `isAtomicSite`  both expect `number | null` whereas `getSite` expects `string | number`.

There may be other ways to address this, but I thought this was more descriptive. Also, in case `''` is passed, then `selectedSite` becomes `null` and `currentPlan` will be `undefined` - which should safely jump over to the Pro plan check.